### PR TITLE
Adds validations to the TeacherProfile model.

### PR DESCRIPTION
**Pull request by ashercodeorg**

### CODE REVIEW

**Comment by ashercodeorg** [comment]
CircleCI failures are unrelated.

File: dashboard/app/models/teacher_profile.rb [hunk 1]
```diff
@@ -20,4 +20,42 @@
...
+  validates_inclusion_of :course, in: COURSES
+
+  YEARS = [
+    YEAR_2013_2014 = '2013-2014'.freeze,
+    YEAR_2014_2015 = '2014-2015'.freeze,
+    YEAR_2015_2016 = '2015-2016'.freeze,
+    YEAR_2016_2017 = '2016-2017'.freeze,
+  ].freeze
+  NO_PD = 'no_pd'.freeze
+  validates_inclusion_of :pd_year, in: YEARS + [NO_PD], allow_nil: true
```
Comment (Hamms) [inline dashboard/app/models/teacher_profile.rb:44]: How much danger does this put us in when the 2017-2018 year hits?

Comment (ashercodeorg) [reply dashboard/app/models/teacher_profile.rb:44]: None. Before using "2017-2018", a developer simply needs to add `YEAR_2017_2018` to  `YEARS`.

Comment (Hamms) [reply dashboard/app/models/teacher_profile.rb:44]: I just mean, given how easy it's going to be to forget that step, how obvious is it going to be that this needs to be done? And how easily might we end up in a situation in which "2017-2018" is available on production but is failing validation?

**Comment by ashercodeorg** [comment]
Given the urgency in populating the table, the constant values were already used from `dashboard-console`. Let me follow-up to see if I should be grabbing the `COURSES` from somewhere else, though I'd prefer to do so in another PR (as necessary). And after this hits production, I plan on manually verifying all existing records pass validation, fixing any that do not.

File: dashboard/app/models/teacher_profile.rb (hunk 1 above)
Comment (ashercodeorg) [reply dashboard/app/models/teacher_profile.rb:44]: Should be pretty obvious: `TeacherProfile::YEAR_2017_2018` isn't defined, running `TeacherProfile.create(..., pd_year: '2017-2018', ...)` will throw an exception, etc.

**Comment by ashercodeorg** [comment]
```
irb(main):001:0> def verify_teacher_profile_valid
irb(main):002:1>   TeacherProfile.all.each do |tp|
irb(main):003:2*     raise unless tp.valid?
irb(main):004:2>   end
irb(main):005:1>   nil
irb(main):006:1> end
=> :verify_teacher_profile_valid
irb(main):007:0> verify_teacher_profile_valid
=> nil
```

**Pull request merged**